### PR TITLE
Add virtio-win registry disk

### PR DESCRIPTION
**Pull request by petrkotas**
**What this PR does / why we need it**:
Package virtio win drivers as a registry disk
to provide easy distribution for kubevirt users

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add virtio win registry disk
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/1720)

### CODE REVIEW

**Comment by fabiand** [comment]
@michalskrivanek can you please doa  review?

File: images/virtio-container-disk/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2017 Red Hat, Inc.
+#
+
+FROM fedora:28 as builder
```
Comment (michalskrivanek) [inline images/virtio-container-disk/Dockerfile]: could we use something more stable? e.g. centos. Or anything small/minimal/stable

File: images/virtio-container-disk/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+RUN curl https://fedorapeople.org/groups/virt/virtio-win/virtio-win.repo -o /etc/yum.repos.d/virtio-win.repo \
+&& dnf install -y virtio-win && mkdir -p /disk 
+
+# the virtio-win package is shipped with version number, move it to standardized location
+# with standardized name
+RUN mv /usr/share/virtio-win/virtio-win-*.iso /disk/virtio-win.iso
```
Comment (michalskrivanek) [inline images/virtio-container-disk/Dockerfile]: what? there is a symlink right there

File: images/virtio-container-disk/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+FROM kubevirt/registry-disk-v1alpha 
+
+LABEL maintainer="The KubeVirt Project <kubevirt-dev@googlegroups.com>"
+
+WORKDIR /disk
+COPY --from=builder /disk/virtio-win.iso ./virtio-win.iso
```
Comment (michalskrivanek) [inline images/virtio-container-disk/Dockerfile]: what about the guest agent binaries?

File: images/virtio-container-disk/Dockerfile (hunk 1 above)
Comment (petrkotas) [reply images/virtio-container-disk/Dockerfile]: Every image we use for upstream is based on fedora. Therefore it make sense to base even this one on Fedora.

Comment (michalskrivanek) [reply images/virtio-container-disk/Dockerfile]: there are 2 F27 already in the code base, F28 will slowly get outdated too. I see in 2 cases "FROM fedora" - would that work? There's nothing specific to a Fedora version here

We should use something better than a random Fedora from the time the code was written originally. But that's a common problem for all the containers in kubevirt, so indeed doesn't have to be solved here

Comment (petrkotas) [reply images/virtio-container-disk/Dockerfile]: Would you kindly please point to specific code locations where do you see it is broken? Thanks in advance.

File: images/virtio-container-disk/Dockerfile (hunk 2 above)
Comment (petrkotas) [reply images/virtio-container-disk/Dockerfile]: Actually good point. Changed it to cp -L

File: images/virtio-container-disk/Dockerfile (hunk 3 above)
Comment (petrkotas) [reply images/virtio-container-disk/Dockerfile]: We do not need them. This image is container disk providing ISO image. ISO contains all binaries required. There are both windows drivers and guest agent install files.

**Comment by petrkotas** [comment]
@slintes @fabiand what do you think? Can we merge?

**Comment by petrkotas** [comment]
I am working on it currently. It cannot be easily tested, since it has to
be done manually.
The best I can think of is mount the disk a and verify the files are there
as designed.
--
s pozdravem,
Petr Kotas

**Comment by fabiand** [comment]
That would be a great start!

**Pull request merged**